qtranslate_core.php

```diff
@@ -621,14 +621,27 @@ function qtranxf_load_option_flag_location( $nm ) {
     }
 }
 
-function qtranxf_validateBool( $var, $default_value ) {
-    if ( $var === '0' ) {
-        return false;
-    } elseif ( $var === '1' ) {
-        return true;
-    } else {
-        return $default_value;
+function qtranxf_validateBool( $var ) {
+    if (is_string( $var )){
+        $val = strtolower( $var );
+        switch ( $val ) {
+            case '0':
+            case 'n':
+            case 'no':
+                return false;
+            case '1':
+            case 'y':
+            case 'yes':
+                return true;
+        }
```

Comment:
This looks better than the previous version, maybe we could keep just this part.
But It could be quite overkill, it's enough to "cast" strings like "1"/"0" with `(bool) val`. I don't know when other cases like `yes`/ `no` or even `y`/`n` are used. Need to check.